README.md

```diff
@@ -1,2 +1,127 @@
-# nbde_client
-Ansible role for configuring Network Bound Disk Encryption clients (e.g. clevis)
+nbde_client
+-----------
+
+Ansible role for configuring Network-Bound Disk Encryption clients (e.g. clevis).
+
+This role currently supports `clevis` as a provider and it uses it for operations like encryption
+and decryption.
+
+
+Supported Distributions
+-----------------------
+* RHEL-7+, CentOS-7+
+* Fedora
+
+
+Limitations
+-----------
+This role can currently create `tang` bindings. TPM2 is not supported as of now.
+
+
+Role Variables
+--------------
+
+These are the variables that can be passed to the role:
+
+
+| **Variable** | **Default/Choices** | **Description** |
+|----------|-------------|------|
+| `nbde_client_provider` | `clevis`| identifies the provider for the `nbde_client` role. We currently support `clevis`.|
+| `nbde_client_bindings` | | a list containing groups of LUKS devices and applicable settings for a binding. This group of devices contains specific binding configurations to be applied to the set of devices.
+|`nbde_client_remote_dir` | `/root/.local/share/nbde_client/` |  specifies a directory in the remote hosts that may be used for storing temporary data such as transferred keyfiles. |
+|`nbde_client_update_initramfs` | `yes` | indicates whether the initramfs should be updated in case changes are made when processing the clevis operations on the devices.|
+
+
+#### nbde_client_bindings
+`nbde_client_bindings` supports the following keys:
+| **Name** | **Default/Choices** | **Description** |
+|----------|-------------|------|
+|`devices` | | specifies LUKS devices, with their path and possibly either a passphrase or key file.|
+|`pin` | | the settings for the binding, like the servers it should bind to. |
```

Comment:
Seems like this parameter is `auth`?  It isn't really a "PIN" or even necessarily a password or pass phrase.  Is this some sort of nomenclature left over from clevis/tang?

Reply:
Yes, it's a clevis nomenclature leftover. I think `auth` would be OK here.